src/common/regions.ts

```diff
@@ -116,27 +121,61 @@ const counties: County[] = chain(state_county_map_dataset)
 class RegionDB {
   constructor(private regions: Region[]) {}
 
-  private filterByType(regionType: RegionType) {
-    return this.all().filter(region => region.regionType === regionType);
-  }
-
   findByFipsCode(fipsCode: FipsCode): Region | null {
     const region = this.regions.find(region => region.fipsCode === fipsCode);
     return region || null;
   }
 
+  findStateByUrlParams(stateId: string): State | null {
+    const foundState = this.states().find(
+      state =>
+        equalLower(state.urlSegment, stateId) ||
+        equalLower(state.stateCode, stateId),
+    );
+    return foundState || null;
+  }
+
+  findCountyByUrlParams(stateId: string, countyId: string): County | null {
+    const foundState = this.findStateByUrlParams(stateId);
+    if (!foundState) {
+      return null;
+    }
+
+    const foundCounty = this.counties()
+      .filter(county => county.state.fipsCode === foundState.fipsCode)
+      .find(county => equalLower(county.urlSegment, countyId));
```

Comment:
nit: Is there a reason to do this as two passes (a filter and then a find) rather than combining?

const foundCounty = this.counties().find(county => 
      county.state.fipsCode === foundState.fipsCode && 
      equalLower(county.urlSegment, countyId)
);

Reply:
I find it a bit cleaner this way, but no strong feelings either way, updating!